test/test_mps.py

```diff
@@ -9227,8 +9227,73 @@ def weight_int8pack_mm(a, b_int8pack, b_scales):
             self.assertLess(mean_err, 0.05)
 
 
-
-
+class TestSDPA(TestCaseMPS):
+    def _compare_tensors(self, y, ref):
+        denom = torch.maximum(ref.abs(), torch.tensor([1e-6], device=ref.device, dtype=ref.dtype))
+        err = ((y - ref).abs() / denom).max().item()
+        self.assertLess(err, 0.01)
+
+    def _test_sdpa_no_mask(self, is_causal: bool):
+        torch.manual_seed(1729)
+        max_seq_length = 72
+        with torch.nn.attention.sdpa_kernel([torch.nn.attention.SDPBackend.MATH]):
+            q = torch.randn([1, 32, 1, 128], dtype=torch.float16, device="mps")
+            k = torch.randn([1, 32, max_seq_length, 128], dtype=q.dtype, device="mps")
+            v = torch.randn([1, 32, max_seq_length, 128], dtype=q.dtype, device="mps")
+                       
+            y, attn = torch.ops.aten._scaled_dot_product_attention_math_for_mps(q, k, v, dropout_p=0.0, is_causal=is_causal)
+            y_ref, attn_ref = torch.ops.aten._scaled_dot_product_attention_math(q, k, v, dropout_p=0.0, is_causal=is_causal)
```

Comment:
Now you can compare against cpu impl as well

Reply:
done!